### DISCUSSION TITLE
Firmware project for Miniware MDP-XP power supply

### DISCUSSION THREAD

**Discussion by violavon** [Ralim/IronOS#1576] [General]
Hello @Ralim (and other interested developers)!

Firstly, thanks for your work on IronOS. I don't know the best way to contact you, so I'm writing here now. You seem to have been involved with Miniware a lot, so this might be of interest to you.

I love Miniware products (especially my TS100), but unfortunately the Miniware MDP-P905, MDP-P906, MDP-M01 or the entire MDP-XP still haven't open-source firmware (I mean something production-ready like IronOS). Do you have any thoughts of this or do you know someone who would be able to and would be interested in working here? I believe there is a demand and we could cooperate with the [OpenDPS project](https://github.com/kanflo/opendps) (check the issue about [this](https://github.com/kanflo/opendps/issues/252)).

**Comment by denruss** [comment]
I would support it, the MDP device is simply excellent, but the FW is so bad that it is terribly inconvenient to use.

**Comment by Ralim** [comment]
I sadly dont currently have the time to start something like this, as much as I would like to.
I'm also a little turned off from these as both of my power supply units have failed over time and need repair more than once (and are currently failed and i haven't fixed them again)
